### ISSUE TITLE
null unmarshalling

### ISSUE THREAD

**Issue by jedborovik** [guregu/null#10]
Unmarshalling and marshalling types in the `null` package isn't symmetric. For example:

```go
type T struct {
  Name null.String
}

in := []byte(`{"Name":""}`)
var t T
json.Unmarshal(in, &t)
fmt.Printf("%+v\n", t) // => {Name:{NullString:{String: Valid:false}}}
out, _ := json.Marshal(t)
fmt.Println(string(out)) // => {"Name":null}
```

Why not have `in` unmarshal to `{Name:{NullString:{String: Valid:true}}}`? In addition to clearing up the issue shown above it would make the `null` package incredibly useful for the common issue of how to determine if a RESTful request intends to set a value to the zero value or if it just wasn't included in the request (a PATCH-like request).

Right now the common recommendation is to use pointers for the fields in your structs as described [here](https://willnorris.com/2014/05/go-rest-apis-and-pointers), but I'd much prefer using the `null` package for this if I could.

Let me know what you think. I'm happy to create a PR for this if you think this idea has legs.

**Comment by guregu** [comment]
Hi, thanks for the report. Blank string input shouldn't produce a null `null.String`... this is inconsistent with the rest of the null package, and the documentation. I think it's a bug, but it's an easy fix! Feel free to send a PR or I will take care of it.

**Comment by guregu** [comment]
Yep, this is an inconsistency I forgot to take care of when I made v2. I have fixed it and bumped the version up to v3. Thanks!

Fixed in a9db3ac26fcd2d70230a8dc6286d0c3694700003.

**Comment by jedborovik** [comment]
You beat me to the PR! Thanks for fixing this.
